source/Octopus.Tentacle/Configuration/Crypto/KubernetesMachineKeyEncryptor.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Kubernetes;
+
+namespace Octopus.Tentacle.Configuration.Crypto
+{
+    public interface IKubernetesMachineKeyEncryptor : IMachineKeyEncryptor
+    {
+    }
+
+public class KubernetesMachineKeyEncryptor : IKubernetesMachineKeyEncryptor
+    {
+        const string SecretName = "tentacle-secret";
+        const string MachineKeyName = "machine-key";
+        const string MachineIvName = "machine-iv";
+
+        readonly byte[]? machineKey;
+        readonly byte[]? machineIv;
+        public KubernetesMachineKeyEncryptor(IKubernetesSecretService kubernetesSecretService, ISystemLog log)
+        {
+            var secret = kubernetesSecretService.TryGet(SecretName, CancellationToken.None).GetAwaiter().GetResult() ?? throw new InvalidOperationException($"Unable to retrieve MachineKey from secret for namespace {KubernetesConfig.Namespace}");
+
+            if (secret.Data is null ||
+                !secret.Data.TryGetValue(MachineKeyName, out machineKey) ||
+                !secret.Data.TryGetValue(MachineIvName, out machineIv))
+            {
+                var (key, iv) = GenerateMachineKey(log);
+                var data = new Dictionary<string, byte[]> { { MachineKeyName, machineKey = key }, { MachineIvName, machineIv = iv } };
+
+                kubernetesSecretService.Patch(SecretName, data, CancellationToken.None).GetAwaiter().GetResult();
+            }
+
+            if (machineKey == null || machineIv == null)
+            {
+                throw new InvalidOperationException("Unable to retrieve or create a machine key for encryption.");
+            }
```

Comment:
I probably would caution against putting service/api calls like this in a constructor.

I'd probably go with something in `Encrypt`/`Decrypt` that makes sure the machine key & IV are correctly loaded before encryption/decryption.

Reply:
Good callout, I realised that I did this in the ConfigMapKeyValueStore as well, so I've updated the other PR so that the configMap is loaded Lazily.